File: data-pipeline/src/paths.ts
// Centralized filesystem paths for the pipeline.
import { fileURLToPath } from 'node:url';
import { dirname, join, resolve } from 'node:path';

const __dirname = dirname(fileURLToPath(import.meta.url));

// data-pipeline/
export const PIPELINE_ROOT = resolve(__dirname, '..');
// laadpalenviewer/
export const PROJECT_ROOT = resolve(PIPELINE_ROOT, '..');

export const CACHE_DIR = join(PIPELINE_ROOT, '.cache');
export const PIPELINE_DATA_DIR = join(PIPELINE_ROOT, 'data');

// webapp output targets
export const WEBAPP_PUBLIC = join(PROJECT_ROOT, 'webapp', 'public');
export const DATA_DIR = join(WEBAPP_PUBLIC, 'data');
export const GEMEENTEN_DIR = join(DATA_DIR, 'gemeenten');
export const MUNICIPALITIES_JSON = join(WEBAPP_PUBLIC, 'municipalities.json');

File: analysis/src/paths.py
"""Filesystem paths for the analysis layer."""
from pathlib import Path

ANALYSIS_ROOT = Path(__file__).resolve().parent.parent
PROJECT_ROOT = ANALYSIS_ROOT.parent

CACHE_DIR = ANALYSIS_ROOT / ".cache"
GOLD_DIR = ANALYSIS_ROOT / "data"  # golden dataset outputs (parquet/csv)

WEBAPP_PUBLIC = PROJECT_ROOT / "webapp" / "public"
DATA_DIR = WEBAPP_PUBLIC / "data"
GEMEENTEN_DIR = DATA_DIR / "gemeenten"
SNAPSHOTS_DIR = DATA_DIR / "snapshots"
DERIVED_DIR = DATA_DIR / "derived"
MUNICIPALITIES_JSON = WEBAPP_PUBLIC / "municipalities.json"

for d in (CACHE_DIR, GOLD_DIR, DERIVED_DIR):
    d.mkdir(parents=True, exist_ok=True)
